cloudinit/config/schema.py

```diff
@@ -148,6 +143,49 @@ def format(self) -> str:
 SchemaProblems = List[SchemaProblem]
 
 
+class SchemaType(Enum):
+    """Supported schema types are etiher cloud-config or network-config.
+
+    Vendordata and Vendordata2 format adheres to cloud-config schema type.
+    Cloud Metadata is unique schema to each cloud platform and likely will not
+    be represented in this enum.
+    """
+
+    CLOUD_CONFIG = "cloud-config"
+    NETWORK_CONFIG = "network-config"
+
+
+# Placeholders for versioned schema and schema file locations.
+# The "latest" key is used in absence of a requested specific version.
+SCHEMA_FILES_BY_TYPE = {
+    SchemaType.CLOUD_CONFIG: {
+        "latest": USERDATA_SCHEMA_FILE,
+    },
+    SchemaType.NETWORK_CONFIG: {
+        "latest": NETWORK_CONFIG_V1_SCHEMA_FILE,
+    },
+}
+
+
+class InstanceDataType(Enum):
+    """Types of instance data provided to cloud-init"""
+
+    USERDATA = "user-data"
+    NETWORK_CONFIG = "network-config"
+    VENDORDATA = "vendor-data"
+    VENDOR2DATA = "vendor2-data"
+    # METADATA = "metadata"
+
+    def __str__(self):  # pylint: disable=invalid-str-returned
```

Comment:
I think we want this class to be a [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) here, right? No need to re-invent it.

Reply:
You are right we do want StrEnum, but we can't yet because we still support python 3.8 on Focal and StrEnum wasn't introduced until 3.11. So until the, we "pre-invent" it :)